org.eclipse.m2e.jdt.tests/src/org/eclipse/m2e/jdt/tests/JpmsConfigurationTest.java

```diff
@@ -0,0 +1,163 @@
+  private static final String JRE_ADD_EXPORTS_VALUE1 = "jdk.jartool/sun.tools.jar=somemodule";
+  private static final String M2E_ADD_OPENS_VALUE1 = M2E_ADD_EXPORTS_VALUE1;
+  private static final String JRE_ADD_OPENS_VALUE1 = JRE_ADD_EXPORTS_VALUE1;
+  private static final String M2E_ADD_READS_VALUE1 = "javafx.graphics=somemodule";
+  private static final String JRE_ADD_READS_VALUE1 = "jdk.jartool=jdk.javadoc";
+  private static final String M2E_PATCH_MODULE_VALUE1 = "javafx.graphics=somepath/some.jar";
+  private static final String JRE_PATCH_MODULE_VALUE1 = "jdk.jartool=somepath/some.jar";
+  
+  private static final String M2E_ADD_EXPORTS_VALUE2 = "javafx.graphics/com.sun.javafx.scene=somemodule";
+  private static final String JRE_ADD_EXPORTS_VALUE2 = "jdk.jartool/sun.security.tools.jarsigner=somemodule";
+  private static final String M2E_ADD_OPENS_VALUE2 = M2E_ADD_EXPORTS_VALUE2;
+  private static final String JRE_ADD_OPENS_VALUE2 = JRE_ADD_EXPORTS_VALUE2;
+  private static final String M2E_ADD_READS_VALUE2 = "javafx.graphics=somemodule2";
+  private static final String JRE_ADD_READS_VALUE2 = "jdk.jartool=java.security.jgss";
+  private static final String M2E_PATCH_MODULE_VALUE2 = "somemodule=somepath/some.jar";
+  private static final String JRE_PATCH_MODULE_VALUE2 = "jdk.javadoc=somepath/some.jar";
+  
+  @Test
+  public void testFileChangeUpdatesJPMSSettings() throws CoreException, IOException, InterruptedException {
+    ((MavenConfigurationImpl) MavenPlugin.getMavenConfiguration()).setAutomaticallyUpdateConfiguration(true);
+    setAutoBuilding(true);
+    
+    File pomFileFS = new File(FileLocator.toFileURL(getClass().getResource("/projects/compilerJpmsSettings/pom.xml")).getFile());
+    File argsSet1FS= new File(FileLocator.toFileURL(getClass().getResource("/projects/compilerJpmsSettings/argsSet1.xml")).getFile());
+    File argsSet2FS = new File(FileLocator.toFileURL(getClass().getResource("/projects/compilerJpmsSettings/argsSet2.xml")).getFile());
+    
+    IProject project = importProject(pomFileFS.getAbsolutePath());
+    waitForJobsToComplete();
+    
+    IJavaProject javaProject = (IJavaProject) project.getNature(JavaCore.NATURE_ID);
+    Map<String, String> jreAttributes = getJreContainerAttributes(javaProject);
+    Map<String, String> m2eAttributes = getM2eContainerAttributes(javaProject);
+    
+    assertEquals("11", javaProject.getOption(JavaCore.COMPILER_SOURCE, false));
+    
+    assertTrue(!jreAttributes.containsKey(ADD_EXPORTS_ATTR));
+    assertTrue(!jreAttributes.containsKey(ADD_OPENS_ATTR));
+    assertTrue(!jreAttributes.containsKey(ADD_READS_ATTR));
+    assertTrue(!jreAttributes.containsKey(PATCH_MODULE_ATTR));
+    
+    assertTrue(!m2eAttributes.containsKey(ADD_EXPORTS_ATTR));
+    assertTrue(!m2eAttributes.containsKey(ADD_OPENS_ATTR));
+    assertTrue(!m2eAttributes.containsKey(ADD_READS_ATTR));
+    assertTrue(!m2eAttributes.containsKey(PATCH_MODULE_ATTR));
```

Comment:
use assertFalse instead, it's easier to read

Reply:
ok done